cafebabel/django_migrations.py

```diff
@@ -1,43 +1,174 @@
-        template=('{prefix} {animation} {percent} ({done}/{total}) '
-                  'ETA: {eta:%H:%M}'),
-        done_char='👤')
+    bar = ProgressBar(total=len(old_users), template=PROGRESSIST_TEMPLATE,
+                      done_char='👤')
     editor_role = Role.objects.get(name='editor')
     with app.app_context():
         for old_user in bar.iter(old_users):
-            data = old_user['fields']
-            creation_date = datetime.strptime(data['date_joined'],
-                                              '%Y-%m-%dT%H:%M:%S')
-            creation_date.replace(tzinfo=timezone.utc)
-            user = ds.create_user(email=data['email'],
-                                  password=data['password'],
-                                  creation_date=creation_date)
-            user.profile.name = f'{data["first_name"]} {data["last_name"]}'
-            avatar = data['avatar']
-            if avatar:
-                user.profile.image_filename = avatar[len('avatars/'):]
-            user.profile.old_pk = old_user['pk']
-            user.save()
-            if data['is_superuser'] or data['is_staff']:
-                ds.add_role_to_user(user, editor_role)
+            create_user(ds, old_user, editor_role)
     click.echo('Users migrated.')
+
+
+def create_user(ds, old_user, editor_role):
+    fields = old_user['fields']
+    creation_date = timestamp_to_datetime(fields['date_joined'])
+    user = ds.create_user(email=fields['email'],
+                          password=fields['password'],
+                          creation_date=creation_date)
+    user.profile.name = f'{fields["first_name"]} {fields["last_name"]}'
+    avatar = fields['avatar']
+    if avatar:
+        user.profile.image_filename = avatar[len('avatars/'):]
+    user.profile.old_pk = old_user['pk']
+    user.save()
+    if fields['is_superuser'] or fields['is_staff']:
+        ds.add_role_to_user(user, editor_role)
+
+
+def migrate_articles(app, limit, article_filepath):
+    click.echo('Starting article migration.')
+    old_articles = load_json_file(article_filepath)
+    if limit:
+        old_articles = old_articles[:limit]
+    bar = ProgressBar(total=len(old_articles), template=PROGRESSIST_TEMPLATE,
+                      done_char='📃')
+    with app.app_context():
+        for old_article in bar.iter(old_articles):
+            create_article(old_article)
+        bar.done = 0  # Reset.
+        for old_article in bar.iter(old_articles):
+            create_article(old_article)
+    click.echo('Articles migrated.')
+
+
+def normalize_status(status):
+    normalized_status = 'draft'
+    if status == 'PUBLISHED':
+        normalized_status = 'published'
+    return normalized_status
+
+
+def normalize_language(language):
+    normalized_language = language[:2]
+    if normalized_language == 'ge':
+        normalized_language = 'de'
+    elif normalized_language == 'sp':
+        normalized_language = 'es'
+    return normalized_language
```

Comment:
Bewarn that _pol_ is in the list of possible languages in the old system config.
Not sure if it's used though.